lib/move_evaluator.rb

```diff
@@ -0,0 +1,85 @@
+class MoveEvaluator
+  def evaluate_move(evaluated_mark, board)
+    @mark = evaluated_mark
+    @board = board
+    alphabeta(evaluated_mark, -1000, 1000)
+  end
+
+  private
+
+  attr_accessor :board
+  attr_accessor :mark
+
+  def alphabeta(current_mark, alpha, beta)
+    if board.win?(switch_mark(mark))
+      return -1
+    elsif board.win?(mark)
+      return 1
+    elsif board.tie?
+      return 0
+    else
+      if current_mark ==mark
+        miniming_player(switch_mark(current_mark), alpha, beta)
+      else
+        maximing_player(switch_mark(current_mark), alpha, beta)
+      end
+    end
+  end
+
+  def switch_mark(mark)
+    if mark == Mark::ROUND
+      Mark::CROSS
+    else
+      Mark::ROUND
+    end
+  end
+
+  def maximing_player(current_mark, alpha, beta)
+    maximum_value = -100
+
+    board.free_positions.each do |position|
+      new_value = get_value_move(current_mark, alpha, beta, position)
+
+      maximum_value = max(maximum_value, new_value)
+      alpha = max(alpha, maximum_value)
```

Comment:
no biggie, but this could be `[alpha, maximum_value].max` if you want to save one private method.